waveform_modes.py

```diff
@@ -315,6 +315,50 @@ def indices(self, ell_m):
             raise ValueError("Input `ell_m` should be an Nx2 sequence of integers")
         return ell_m[:, 0] * (ell_m[:, 0] + 1) - self.ell_min ** 2 + ell_m[:, 1]
 
+    def apply_eth(self, N_eth=0, N_ethbar=0, eth_convention='NP'):
```

Comment:
Since this function signature enables it, and it is sometimes useful to consecutively apply various operators, I wonder if it would be better to use some way of specifying the order of operators.  For example, I'm sure I've seen eth(eth(ethbar(ethbar(f)))) [or the conjugate], and I'm pretty sure I've sure eth(ethbar(eth(ethbar(f)))).  So maybe instead of `N_eth` and `N_ethbar`, there could be an argument like `operations` that takes a string of `'+'` for raise and `'-'` for lower.  Then, my first example would be `'++--'` and my second would be `'+-+-'`.  What do you think?

Reply:
Ahh that's a great idea! that'll be much for flexible, I'll implement this.

Reply:
I just noticed a little ambiguity here regarding the order of operations.  Obviously, eth should be '+' and ethbar should be '-'.  I think what I wrote is the most appropriate choice, so for example eth(ethbar(f)) would be written as '+-'.  But note that this means applying ethbar *first*, which means iterating through the string in *reverse* order:
```python
for operation in reversed(operations):
```
Of course, people might expect the opposite order, so this should be emphasized in the docstring.

It's also probably best to check for valid operations before starting that loop.  Here's a simple way to do it:
```python
if not set(operations).issubset({'+', '-'}):
    raise ValueError("Operations can be combinations of '+' and '-'; got '{0}'".format(operations)
```